src/shared/json.c

```diff
@@ -1768,6 +1768,7 @@ static int json_format(FILE *f, JsonVariant *v, JsonFormatFlags flags, const cha
 }
 
 int json_variant_format(JsonVariant *v, JsonFormatFlags flags, char **ret) {
+        _cleanup_fclose_ FILE *f = NULL;
```

Comment:
`f` needs to be closed before freeing `s`.

Reply:
Whoops, indeed, thank you.